### ISSUE TITLE
All WeakAuras have been wiped

### ISSUE THREAD

**Issue by jakeisonline** [WeakAuras/WeakAuras2#320]
### Unable to reproduce

This happened seemingly randomly. It did not coincide with an update. It *does* seem to coincide with a minor change to a label on an aura. From there I simply signed in, and they're gone.

#### What version of WeakAuras are you using?

2.4.25

#### Do you have an error log of what happened?

I was able to pull the original WA file from one of the `.bak` files. I have attached those here: 
[SavedVariables.zip](https://github.com/WeakAuras/WeakAuras2/files/1568872/SavedVariables.zip)

Upon adding this into the main `WeakAuras.lua` I get the following:

````
WTF\Account\101274380#1\SavedVariables\WeakAuras-2.4.25.lua:35271: unexpected symbol near '<eof>'
````

#### Please provide any additional information below.

The `WeakAuras.lua` will reset itself to the following:

````

WeakAurasSaved = {
	["dynamicIconCache"] = {
	},
	["login_squelch_time"] = 10,
	["registered"] = {
	},
	["displays"] = {
	},
	["editor_theme"] = "Monokai",
}
````

It would appear there is somehow a syntax error in the lua file, which causes a panic during initialization?

**Comment by jakeisonline** [comment]
`WeakAuras-2.4.25.lua:35271` does seem to end abruptly. Is there a length limit on these files?

**Comment by InfusOnWoW** [comment]
Not that we are aware of. And it's wow that's saving those files, we have zero control over the actual settings saving.

Are you running any anti virus software?

**Comment by jakeisonline** [comment]
Bitdefender is running yeah. Come to think of it, pretty sure I had to force close wow very recently... could explain the unfinished file.

**Comment by InfusOnWoW** [comment]
The last time lots of people had problems, it was due to avast preventing wow from properly saving the settings. Anyway, there's nothing we can do if wow doesn't write the file completly. 

Can you check whether the WeakAuras.lua.bak file in WTF/Account/AccountName/SavedVariables contains anything?

**Comment by jakeisonline** [comment]
After noticing the problem I looked at the `.bak` file. `WeakAuras.lua.bak` contained content, which I copied into `WeakAuras.lua` (this is reflected in the `.zip` attached earlier). Keep in mind I may have run wow several times before noticing a problem with WA due to my sparse use of it on alts.

My knowledge of how wow writes settings files is close to zero. I assume, as all my settings files have varied modified dates, wow only updates a file if an addon's settings have changed during that play session. It also only seems to write the file at the end of a play session? If I have this loosely right then I think this is a non-issue.

I've been having intermittent internet connection and windows issues. I think an unfortunate set of circumstances triggered by my force shutting down of wow and/or my PC caused a file to be half written.

Thanks for taking a look anyway, sorry to have wasted your time. I can't live without this addon!
